client/service/src/builder.rs

```diff
@@ -429,7 +429,7 @@ impl<TBl, TRtApi, TCl, TFchr, TSc, TImpQu, TFprb, TFpp, TExPool, TRpc, Backend>
 	/// Defines which import queue to use.
 	pub fn with_import_queue<UImpQu>(
 		self,
-		builder: impl FnOnce(&Configuration, Arc<TCl>, Option<TSc>, Arc<TExPool>)
+		builder: impl FnOnce(&Configuration, Arc<TCl>, Option<TSc>, Arc<TExPool>, &TaskManager)
```

Comment:
I really think that we should not expose the `TaskManager` in the public API. It's an implementation detail.
Please put back a `SpawnTaskHandle`.

Reply:
Haha all right I was about to ask you in this PR if that was supposed to be like this! :)